packages/flutter_tools/lib/src/application_package.dart

```diff
@@ -429,8 +443,28 @@ class ApkManifestData {
     _Element launchActivity;
     for (_Element activity in activities) {
       final _Attribute enabled = activity.firstAttribute('android:enabled');
-      if (enabled == null || enabled.value.contains('0xffffffff')) {
-        launchActivity = activity;
+      final Iterable<_Element> intentFilters =
```

Comment:
Isn't this logic the same as what was already implemented up above?  If so, we shouldn't replicate the same behavior twice.  Please extract a function that figures out which Activity we should launch with.